File: Deep-Learning-for-Beginners-master/ch.5/Dropout.m
function ym = Dropout(y, ratio)
  [m, n] = size(y);  
  ym     = zeros(m, n);
  
  num     = round(m*n*(1-ratio));
  idx     = randperm(m*n, num);
  ym(idx) = 1 / (1-ratio);
end

File: Deep-Learning-for-Beginners-master/Python/ch.5/Dropout.py
import numpy as np


def Dropout(y, ratio):
    ym = np.zeros_like(y)
    
    num = round(y.size*(1-ratio))
    idx = np.random.choice(y.size, num, replace=False)
    ym[idx] = 1.0 / (1.0 - ratio)
    
    return ym